extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -61,30 +63,87 @@ export const mediaMimes = new Set([
 	'audio/x-wav',
 ]);
 
-export async function getMarkdownLink(document: vscode.TextDocument, ranges: readonly vscode.Range[], urlList: string, token: vscode.CancellationToken): Promise<{ additionalEdits: vscode.WorkspaceEdit; label: string } | undefined> {
+const smartPasteRegexes = [
+	{ regex: /\[.*\]\(.*\)/g, is_markdown_link: true }, // Is a Markdown Link
+	{ regex: /!\[.*\]\(.*\)/g, is_markdown_link: true }, // Is a Markdown Image Link
+	{ regex: /\[([^\]]*)\]\(([^)]*)\)/g, is_markdown_link: false }, // In a Markdown link
+	{ regex: /^```[\s\S]*?```$/gm, is_markdown_link: false }, // In a fenced code block
+	{ regex: /^\$\$[\s\S]*?\$\$$/gm, is_markdown_link: false }, // In a fenced math block
+	{ regex: /`[^`]*`/g, is_markdown_link: false }, // In inline code
+	{ regex: /\$[^$]*\$/g, is_markdown_link: false }, // In inline math
+];
+
+export interface SkinnyTextDocument extends vscode.TextDocument {
```

Comment:
`extends vscode.TextDocument` is actually reverse of what you want. Instead `SkinnyTextDocument ` should be a subtype of `vscode.TextDocument`. Simply removing the `extends vscode.TextDocument` should fix this